diff --git a/codegen/js/JavascriptTarget.js b/codegen/js/JavascriptTarget.js
index a5dadb70e9ff8a1358b26415efdc5db139c30016..54e0e1e6498f0ad551f7bc653a64bc67d46f39a8 100644
--- a/codegen/js/JavascriptTarget.js
+++ b/codegen/js/JavascriptTarget.js
@@ -173,11 +173,14 @@ function createJavascriptTarget(exposeNames) {
         }
         ;
         getStateTypeInfo(state) {
+            console.log('getStateTypeInfo', state);
             const typeNames = new Set();
             const toSerializableType = (type) => {
+                console.log('toSerializableType', type);
                 if (type.kind !== KnownType_1.TypeKind.Simple) {
                     throw new Error(`Non-serializable type in state: ${(0, KnownType_1.stringifyType)(type)}`);
                 }
+                console.log('toSerializableType: adding', type.name);
                 typeNames.add(type.name);
                 const typeInfo = this.typechecker.fetchTypeInfoFromSimpleType(type);
                 if (typeInfo?.attributes.some(x => x.kind === Attribute_1.TypeAttributeKind.IsUserDefined)) {
@@ -190,6 +193,7 @@ function createJavascriptTarget(exposeNames) {
                         }
                     }
                 }
+                console.log('toSerializableType: returning');
                 return {
                     name: type.name,
                     typeArguments: type.typeArguments.map(toSerializableType),
@@ -199,13 +203,16 @@ function createJavascriptTarget(exposeNames) {
                 st.name,
                 toSerializableType(this.typechecker.fetchVariableDefinitionType(st))
             ]);
+            console.log('getStateTypeInfo typeNamesInState', typeNames, [...typeNames]);
             return {
                 typeNamesInState: [...typeNames],
                 nameTypePairs,
             };
         }
         generateSerializer({ nameTypePairs, typeNamesInState }) {
+            console.log('generateSerializer typeNamesInState', typeNamesInState);
             return `state=>{let id=0;const objs=[];const refMap=new Map();const $serialize=(obj,type)=>{if(refMap.has(obj))return refMap.get(obj);const myId=id++;refMap.set(obj,myId);objs[myId]={${joinBy(',', typeNamesInState, name => {
+                console.log('generateSerializer for', name);
                 let serializer;
                 if (Primitives_1.primitiveTypes.some(x => x.name === name)) {
                     switch (name) {
@@ -236,6 +243,7 @@ function createJavascriptTarget(exposeNames) {
                         serializer = this.impl.types[name](this.getBuiltinVariableName).serialize;
                     }
                 }
+                console.log('generateSerializer returning', `${JSON.stringify(name)}:${serializer}`);
                 return `${JSON.stringify(name)}:${serializer}`;
             })}}[type.name](obj,type,$serialize);return myId};const refs={${joinBy(',', nameTypePairs, ([id, type]) => {
                 const stateName = this.getPublicIdentifierString(id);
